query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/fields/LdapAttributeName.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.ldap.fields;
+
+import static org.codice.ddf.admin.ldap.commons.LdapMessages.invalidUserAttribute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.codice.ddf.admin.api.report.ErrorMessage;
+import org.codice.ddf.admin.common.fields.base.scalar.StringField;
+
+public class LdapAttributeName extends StringField {
+
+    public static final String DEFAULT_FIELD_NAME = "attriName";
+
+    public static final String FIELD_TYPE_NAME = "LdapAttributeName";
+
+    public static final String DESCRIPTION =
+            "The short descriptive name of an LDAP attribute as defined in RFC2251.";
+
+    private static final String REGEX = "^[a-zA-Z0-9]+[a-zA-Z0-9-]*$";
+
+    public LdapAttributeName() {
+        super(DEFAULT_FIELD_NAME, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    public LdapAttributeName(String fieldName) {
+        super(fieldName, FIELD_TYPE_NAME, DESCRIPTION);
+    }
+
+    @Override
+    public List<ErrorMessage> validate() {
+        List<ErrorMessage> errors = super.validate();
+        if (!errors.isEmpty()) {
+            return errors;
+        }
+
+        if(getValue() != null) {
+            errors.addAll(validate(getValue(), path()));
+        }
+        return errors;
+    }
+
+    // TODO: 7/7/17 - tbatie - This validate should be reformatted once there is a generic way to create MapField objects that contain different value field.
+    public static List<ErrorMessage> validate(String attribute, List<String> path) {
+        List<ErrorMessage> errors = new ArrayList<>();
+
+        // https://docs.ldap.com/specs/rfc1778.txt  : RFC 1778 section 2 and 2.15
```

Comment:
That's definitely not the right RFC. That refers to the _values_ that can be in attributes. E.g. `cn: Blen Desta` where `Blen Desta` is the value of the `cn` attribute.